configure.ac

```diff
@@ -1807,6 +1807,48 @@ AC_LINK_IFELSE([
     AC_MSG_RESULT([no])
 ])
 
+# Linux pthread_getname_np:
+AC_MSG_CHECKING([pthread_getname_np()])
+AC_LINK_IFELSE([
+    AC_LANG_PROGRAM([
+#if HAVE_PTHREAD_H
+#include <pthread.h>
+#endif
+#if PTHREAD_NP_H
+#include <pthread_np.h>
+#endif
+      ], [
+        char name[[32]];
+        pthread_getname_np(pthread_self(), name, sizeof(name));
+    ])
+  ], [
+    AC_DEFINE(HAVE_PTHREAD_GETNAME_NP, 1, [Whether pthread_getname_np() is available])
+    AC_MSG_RESULT([yes])
+  ], [
+    AC_MSG_RESULT([no])
+])
+
+# BSD pthread_getname_np:
```

Comment:
Not a big deal, but I'm wondering if we should make more of an effort to not do tests which we know are going to fail? In this case, if the previous check has succeeded, we know that this one here (for BSD) can (and should!) not succeed, so why bother compiling these test cases?

Reply:
Combined the two checks.